File: CS/WpfApplication/ColorHelper/BindingToColorConverter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Data;
using System.Windows.Documents;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Imaging;
using System.Windows.Navigation;
using System.Windows.Shapes;
using System.Windows.Markup;
using System.Collections.ObjectModel;
using DevExpress.Xpf.Grid;

namespace WpfApplication
{
    public class BindingToColorConverter : DependencyObject, IMultiValueConverter
    {


        object IMultiValueConverter.Convert(object[] values, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            ObservableCollection<HighlightedGridCell> cellsToHighlight = values[0] as ObservableCollection<HighlightedGridCell>;
            object row = values[1];
            object column = values[2];
            return GetColorToHighlight(row, column, cellsToHighlight);
        }

        object[] IMultiValueConverter.ConvertBack(object value, Type[] targetTypes, object parameter, System.Globalization.CultureInfo culture)
        {
            throw new NotImplementedException();
        }
        private object GetColorToHighlight(object row, object column, ObservableCollection<HighlightedGridCell> cellsToHighlight)
        {
            if (row == null || column == null)
                return null;
            foreach (HighlightedGridCell cell in cellsToHighlight)
            {
                if (cell.Column == column && cell.Row == row)
                    return new SolidColorBrush(cell.Color);
            }
            return null;
        }

     
    }
}

File: VB/WpfApplication/ColorHelper/BindingToColorConverter.vb
Imports System
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Data
Imports System.Windows.Media
Imports System.Collections.ObjectModel

Namespace WpfApplication

    Public Class BindingToColorConverter
        Inherits DependencyObject
        Implements IMultiValueConverter

        Private Function Convert(ByVal values As Object(), ByVal targetType As Type, ByVal parameter As Object, ByVal culture As Globalization.CultureInfo) As Object Implements IMultiValueConverter.Convert
            Dim cellsToHighlight As ObservableCollection(Of HighlightedGridCell) = TryCast(values(0), ObservableCollection(Of HighlightedGridCell))
            Dim row As Object = values(1)
            Dim column As Object = values(2)
            Return GetColorToHighlight(row, column, cellsToHighlight)
        End Function

        Private Function ConvertBack(ByVal value As Object, ByVal targetTypes As Type(), ByVal parameter As Object, ByVal culture As Globalization.CultureInfo) As Object() Implements IMultiValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function

        Private Function GetColorToHighlight(ByVal row As Object, ByVal column As Object, ByVal cellsToHighlight As ObservableCollection(Of HighlightedGridCell)) As Object
            If row Is Nothing OrElse column Is Nothing Then Return Nothing
            For Each cell As HighlightedGridCell In cellsToHighlight
                If cell.Column Is column AndAlso cell.Row Is row Then Return New SolidColorBrush(cell.Color)
            Next

            Return Nothing
        End Function
    End Class
End Namespace
